blaze-server/src/main/scala/org/http4s/server/blaze/BlazeServerBuilder.scala

```diff
@@ -324,22 +327,24 @@ class BlazeServerBuilder[F[_]] private (
         dispatcher
       )
 
-    Future.successful {
-      engineConfig match {
-        case Some((ctx, configure)) =>
-          val engine = ctx.createSSLEngine()
-          engine.setUseClientMode(false)
-          configure(engine)
-
-          LeafBuilder(
-            if (isHttp2Enabled) http2Stage(engine)
-            else http1Stage(secure = true, engine.some)
-          ).prepend(new SSLStage(engine))
-
-        case None =>
-          if (isHttp2Enabled)
-            logger.warn("HTTP/2 support requires TLS. Falling back to HTTP/1.")
-          LeafBuilder(http1Stage(secure = false, None))
+    dispatcher.unsafeToFuture {
```

Comment:
Alternatively, we could construct the EC around the place where we sequence `Dispatcher` (`dispatcher <- Resource.eval(Dispatcher[F].allocated.map(_._1))`)